tests/components/test_logbook.py

```diff
@@ -558,6 +558,87 @@ def create_state_changed_event(self, event_time_fired, entity_id, state,
     assert response.status == 200
 
 
+async def test_logbook_view_period_entity(hass, aiohttp_client):
+    """Test the logbook view with period and entity."""
+    await hass.async_add_job(init_recorder_component, hass)
+    await async_setup_component(hass, 'logbook', {})
+    await hass.components.recorder.wait_connection_ready()
+    await hass.async_add_job(hass.data[recorder.DATA_INSTANCE].block_till_done)
+
+    entity_id = 'switch.test'
+    hass.states.async_set(entity_id, STATE_OFF)
+    hass.states.async_set(entity_id, STATE_ON)
+    await hass.async_block_till_done()
+
+    client = await aiohttp_client(hass.http.app)
+
+    # Today time 00:00:00
+    start = dt_util.utcnow().date()
+    start_date = datetime(start.year, start.month, start.day)
+
+    # Test today entries without filters
+    response = await client.get(
+        '/api/logbook/{}'.format(start_date.isoformat()))
+    assert response.status == 200
+    json = await response.json()
+    assert len(json) == 1
+    assert json[0]['entity_id'] == entity_id
+
+    # Test today entries with filter by period
+    response = await client.get(
+        '/api/logbook/{}?period=1'.format(start_date.isoformat()))
+    assert response.status == 200
+    json = await response.json()
+    assert len(json) == 1
+    assert json[0]['entity_id'] == entity_id
+
+    # Test today entries with filter by entity_id
+    response = await client.get(
+        '/api/logbook/{}?entity=switch.test'.format(
+            start_date.isoformat()))
+    assert response.status == 200
+    json = await response.json()
+    assert len(json) == 1
+    assert json[0]['entity_id'] == entity_id
+
+    # Test entries for 3 days with filter by entity_id
+    response = await client.get(
+        '/api/logbook/{}?period=3&entity=switch.test'.format(
+            start_date.isoformat()))
+    assert response.status == 200
+    json = await response.json()
+    assert len(json) == 1
```

Comment:
To test this, you will need to add states that are stored outside of the period. You then will have to check that the number of returned rows is less because some states are no longer there.

Reply:
I do it below where period changes